Does not compile on kernel 5.0

DKMS make.log for mtcadummy-0.9.2 for kernel 5.0.0-15-generic (x86_64)
Fr 24. Mai 00:05:11 CEST 2019
make -C /lib/modules/5.0.0-15-generic/build V=1 M=/var/lib/dkms/mtcadummy/0.9.2/build modules
make[1]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
make[1]: Entering directory '/usr/src/linux-headers-5.0.0-15-generic'
test -e include/generated/autoconf.h -a -e include/config/auto.conf || (                \
echo >&2;                                                       \
echo >&2 "  ERROR: Kernel configuration is invalid.";           \
echo >&2 "         include/generated/autoconf.h or include/config/auto.conf are missing.";\
echo >&2 "         Run 'make oldconfig && make prepare' on kernel src to fix it.";      \
echo >&2 ;                                                      \
/bin/false)
mkdir -p /var/lib/dkms/mtcadummy/0.9.2/build/.tmp_versions ; rm -f /var/lib/dkms/mtcadummy/0.9.2/build/.tmp_versions/*
make -f ./scripts/Makefile.build obj=/var/lib/dkms/mtcadummy/0.9.2/build
  gcc -Wp,-MD,/var/lib/dkms/mtcadummy/0.9.2/build/.mtcadummy_drv.o.d  -nostdinc -isystem /usr/lib/gcc/x86_64-linux-gnu/8/include  -I./arch/x86/include -I./arch/x86/include/generated  -I./include -I./arch/x86/include/uapi -I./arch/x86/include/generated/uapi -I./include/uapi -I./include/generated/uapi -include ./include/linux/kconfig.h -Iubuntu/include  -include ./include/linux/compiler_types.h -D__KERNEL__ -Wall -Wundef -Werror=strict-prototypes -Wno-trigraphs -fno-strict-aliasing -fno-common -fshort-wchar -fno-PIE -Werror-implicit-function-declaration -Werror=implicit-int -Wno-format-security -std=gnu89 -mno-sse -mno-mmx -mno-sse2 -mno-3dnow -mno-avx -m64 -falign-jumps=1 -falign-loops=1 -mno-80387 -mno-fp-ret-in-387 -mpreferred-stack-boundary=3 -mskip-rax-setup -mtune=generic -mno-red-zone -mcmodel=kernel -DCONFIG_X86_X32_ABI -DCONFIG_AS_CFI=1 -DCONFIG_AS_CFI_SIGNAL_FRAME=1 -DCONFIG_AS_CFI_SECTIONS=1 -DCONFIG_AS_FXSAVEQ=1 -DCONFIG_AS_SSSE3=1 -DCONFIG_AS_AVX=1 -DCONFIG_AS_AVX2=1 -DCONFIG_AS_AVX512=1 -DCONFIG_AS_SHA1_NI=1 -DCONFIG_AS_SHA256_NI=1 -Wno-sign-compare -fno-asynchronous-unwind-tables -mindirect-branch=thunk-extern -mindirect-branch-register -fno-delete-null-pointer-checks -Wno-frame-address -Wno-format-truncation -Wno-format-overflow -Wno-int-in-bool-context -O2 --param=allow-store-data-races=0 -Wframe-larger-than=1024 -fstack-protector-strong -Wno-unused-but-set-variable -Wno-unused-const-variable -fno-omit-frame-pointer -fno-optimize-sibling-calls -fno-var-tracking-assignments -pg -mrecord-mcount -mfentry -DCC_USING_FENTRY -Wdeclaration-after-statement -Wvla -Wno-pointer-sign -Wno-stringop-truncation -fno-strict-overflow -fno-merge-all-constants -fmerge-constants -fno-stack-check -fconserve-stack -Werror=date-time -Werror=incompatible-pointer-types -Werror=designated-init -fmacro-prefix-map=./= -Wno-packed-not-aligned  -DMODULE  -DKBUILD_BASENAME='"mtcadummy_drv"' -DKBUILD_MODNAME='"mtcadummy"' -c -o /var/lib/dkms/mtcadummy/0.9.2/build/mtcadummy_drv.o /var/lib/dkms/mtcadummy/0.9.2/build/mtcadummy_drv.c
/var/lib/dkms/mtcadummy/0.9.2/build/mtcadummy_drv.c: In function ‘mtcaDummy_ioctl’:
/var/lib/dkms/mtcadummy/0.9.2/build/mtcadummy_drv.c:417:72: error: macro "access_ok" passed 3 arguments, but takes just 2
     status = !access_ok(VERIFY_WRITE, (void __user*)arg, _IOC_SIZE(cmd));
                                                                        ^
/var/lib/dkms/mtcadummy/0.9.2/build/mtcadummy_drv.c:417:15: error: ‘access_ok’ undeclared (first use in this function)
     status = !access_ok(VERIFY_WRITE, (void __user*)arg, _IOC_SIZE(cmd));
               ^~~~~~~~~
/var/lib/dkms/mtcadummy/0.9.2/build/mtcadummy_drv.c:417:15: note: each undeclared identifier is reported only once for each function it appears in
/var/lib/dkms/mtcadummy/0.9.2/build/mtcadummy_drv.c:419:71: error: macro "access_ok" passed 3 arguments, but takes just 2
     status = !access_ok(VERIFY_READ, (void __user*)arg, _IOC_SIZE(cmd));
                                                                       ^
/var/lib/dkms/mtcadummy/0.9.2/build/mtcadummy_drv.c: In function ‘llrfDummy_ioctl’:
/var/lib/dkms/mtcadummy/0.9.2/build/mtcadummy_drv.c:514:72: error: macro "access_ok" passed 3 arguments, but takes just 2
     status = !access_ok(VERIFY_WRITE, (void __user*)arg, _IOC_SIZE(cmd));
                                                                        ^
/var/lib/dkms/mtcadummy/0.9.2/build/mtcadummy_drv.c:514:15: error: ‘access_ok’ undeclared (first use in this function)
     status = !access_ok(VERIFY_WRITE, (void __user*)arg, _IOC_SIZE(cmd));
               ^~~~~~~~~
/var/lib/dkms/mtcadummy/0.9.2/build/mtcadummy_drv.c:516:71: error: macro "access_ok" passed 3 arguments, but takes just 2
     status = !access_ok(VERIFY_READ, (void __user*)arg, _IOC_SIZE(cmd));
                                                                       ^
/var/lib/dkms/mtcadummy/0.9.2/build/mtcadummy_drv.c: In function ‘pcieuniDummy_ioctl’:
/var/lib/dkms/mtcadummy/0.9.2/build/mtcadummy_drv.c:584:72: error: macro "access_ok" passed 3 arguments, but takes just 2
     status = !access_ok(VERIFY_WRITE, (void __user*)arg, _IOC_SIZE(cmd));
                                                                        ^
/var/lib/dkms/mtcadummy/0.9.2/build/mtcadummy_drv.c:584:15: error: ‘access_ok’ undeclared (first use in this function)
     status = !access_ok(VERIFY_WRITE, (void __user*)arg, _IOC_SIZE(cmd));
               ^~~~~~~~~
/var/lib/dkms/mtcadummy/0.9.2/build/mtcadummy_drv.c:586:71: error: macro "access_ok" passed 3 arguments, but takes just 2
     status = !access_ok(VERIFY_READ, (void __user*)arg, _IOC_SIZE(cmd));
                                                                       ^
make[2]: *** [scripts/Makefile.build:286: /var/lib/dkms/mtcadummy/0.9.2/build/mtcadummy_drv.o] Error 1
make[1]: *** [Makefile:1584: _module_/var/lib/dkms/mtcadummy/0.9.2/build] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-5.0.0-15-generic'
make: *** [Makefile:9: all] Error 2

--- a/mtcadummy_drv.c
+++ b/mtcadummy_drv.c
@@ -413,10 +413,14 @@ static long mtcaDummy_ioctl(struct file* filp, unsigned int cmd, unsigned long a
     return -ENOTTY;
   }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,0,0)
   if(_IOC_DIR(cmd) & _IOC_READ)
     status = !access_ok(VERIFY_WRITE, (void __user*)arg, _IOC_SIZE(cmd));
   else if(_IOC_DIR(cmd) & _IOC_WRITE)
     status = !access_ok(VERIFY_READ, (void __user*)arg, _IOC_SIZE(cmd));
+#else
+  status = !access_ok((void __user*)arg, _IOC_SIZE(cmd));
+#endif
   if(status) {
     dbg_print("Incorrect ioctl command %d\n", cmd);
     return -EFAULT;
@@ -510,10 +514,14 @@ static long llrfDummy_ioctl(struct file* filp, unsigned int cmd, unsigned long a
     return -ENOTTY;
   }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,0,0)
   if(_IOC_DIR(cmd) & _IOC_READ)
     status = !access_ok(VERIFY_WRITE, (void __user*)arg, _IOC_SIZE(cmd));
   else if(_IOC_DIR(cmd) & _IOC_WRITE)
     status = !access_ok(VERIFY_READ, (void __user*)arg, _IOC_SIZE(cmd));
+#else
+  status = !access_ok((void __user*)arg, _IOC_SIZE(cmd));
+#endif
   if(status) {
     dbg_print("Incorrect ioctl command %d\n", cmd);
     return -EFAULT;
@@ -580,10 +588,14 @@ static long pcieuniDummy_ioctl(struct file* filp, unsigned int cmd, unsigned lon
     return -ENOTTY;
   }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,0,0)
   if(_IOC_DIR(cmd) & _IOC_READ)
     status = !access_ok(VERIFY_WRITE, (void __user*)arg, _IOC_SIZE(cmd));
   else if(_IOC_DIR(cmd) & _IOC_WRITE)
     status = !access_ok(VERIFY_READ, (void __user*)arg, _IOC_SIZE(cmd));
+#else
+  status = !access_ok((void __user*)arg, _IOC_SIZE(cmd));
+#endif
   if(status) {
     dbg_print("Incorrect ioctl command %d\n", cmd);
     return -EFAULT;

